cmd/gaia/app/sim_test.go

```diff
@@ -67,13 +67,40 @@ func appStateFn(r *rand.Rand, accs []simulation.Account) json.RawMessage {
 		"\t{amount of steak per account: %v, initially bonded validators: %v}\n",
 		amount, numInitiallyBonded)
 
-	// Randomly generate some genesis accounts
-	for _, acc := range accs {
+	// randomly generate some genesis accounts
+	for i, acc := range accs {
 		coins := sdk.Coins{sdk.NewCoin(stakingTypes.DefaultBondDenom, sdk.NewInt(amount))}
-		genesisAccounts = append(genesisAccounts, GenesisAccount{
-			Address: acc.Address,
-			Coins:   coins,
-		})
+		bacc := auth.NewBaseAccountWithAddress(acc.Address)
+		bacc.SetCoins(coins)
+
+		var gacc GenesisAccount
+
+		// create a vesting account under the following conditions:
+		// - not an initially bonded (genesis) validator
+		// - every 10th account past the initial bonded validator set
+		if int64(i) > numInitiallyBonded && i%10 == 0 {
+			var vacc auth.VestingAccount
+
+			// Vesting starts at the genesis time and ends at a time between genesis
+			// and two days from genesis.
+			//
+			// TODO: Do the vesting periods need to be shrunk to account for short
+			// simulations?
+			startTime := genesisTimestamp.Unix()
+			endTime := r.Int63n((startTime+(60*60*24*2))-startTime) + startTime
```

Comment:
@cwgoes is the end time generated here reasonable? Or should it be much shorter...in the manner of seconds?

Reply:
I think this is OK - a few-hundred block sim will run several days in virtual time.

Randomize it some more though - some vesting accounts should vest very quickly; others very slowly.